packages/core/strapi/lib/core/domain/content-type/validator.js

```diff
@@ -24,6 +24,9 @@ const LIFECYCLES = [
   'afterDeleteMany',
 ];
 
+// See GraphQL Spec https://spec.graphql.org/June2018/#sec-Names
+const GRAPHQL_ENUM_REGEX = new RegExp('^[_A-Za-z][_0-9A-Za-z]*$');
```

Comment:
Maybe it's better if we're talking about the enum regexp instead?
I would prefer we avoid coupling/referencing GraphQL to core code

Reply:
I agree, but on the other hand the GraphQL is the reason why need to this. Therefore I'd prefer to explicit here and to explain why ...

Reply:
Added in f880614ad8b69d0f6cd2be4192acea571e14bc4d Does that work for you?